contracts/FTokenStrat.sol

```diff
@@ -52,21 +56,31 @@ contract FTokenStrat is IStrat {
         _;
     }
 
-    constructor(IVault vault_, IFToken fToken_) {
+    constructor(IVault vault_, IFToken fToken_, IRewardPool rewardpool_) {
         require(address(vault_.underlying()) == fToken_.underlying(),"VAULT / TOKEN UNDERLYING MISMATCH");
         vault = vault_;
         fToken = fToken_;
+        rewardpool = rewardpool_;
+        rewardtoken = IERC20Detailed(rewardpool.rewardToken());
         timelock = Timelock(vault.timelock()); // use the same timelock from the vault
         underlying = IERC20Detailed(fToken_.underlying());
         underlying.safeApprove(address(fToken), uint(-1)); // intentional underflow
+        fToken.approve(address(rewardpool), uint(-1));
+
         minWithdrawalCap = 1000 * (10 ** underlying.decimals()); // 10k min withdrawal cap
     }
 
     function invest() external override onlyVault {
         uint balance = underlying.balanceOf(address(this));
+
         if(balance > buffer) {
-            fToken.deposit(balance - buffer); // can't underflow because of above if statement
+            fToken.deposit(balance - buffer);
+
+            uint ftokenBalance = fToken.underlyingBalanceWithInvestmentForHolder(address(this));
```

Comment:
This seems like the underlying (e.g. Dai) balance, not fDai balance. I think in this case we need to just do `fToken.balanceOf(address(this))` instead